### PR TITLE
[Configuration] Remove the need for maintaining an on-chain config registry

### PR DESCRIPTION
**Pull request by sitalkedia**
### Description

See https://github.com/aptos-labs/aptos-core/issues/9150 for more details

### Test Plan

Existing Unit tests.

Tested it by running `test_onchain_shuffling_change`, which toggles the shuffling flag and ensures that the txn shuffling actually takes place.

### CODE REVIEW

**Review by zekun000 — COMMENTED** [review]
looks reasonable to me, cc @JoshLind to take a look too

File: vm-validator/Cargo.toml
Comment (JoshLind) [inline vm-validator/Cargo.toml:17]: Silly question: do we need to add these? They look unused below, right?

File: state-sync/inter-component/event-notifications/src/lib.rs
Comment (JoshLind) [inline state-sync/inter-component/event-notifications/src/lib.rs:380]: Can we avoid putting this here? Feels like it would be better in `types/src/on_chain_config/mod.rs`, no?

Also, it would avoid the need for us to import `anyhow` into state sync (which I've tried to avoid because we have our own notion of error and result). 😄

**Review by JoshLind — COMMENTED** [review]
Thanks @sitalkedia. LGTM -- only concern is the location of `DbBackedOnChainConfig`.

File: state-sync/inter-component/event-notifications/src/lib.rs
Comment (sitalkedia) [reply state-sync/inter-component/event-notifications/src/lib.rs:380]: @JoshLind - We can't put the `DbBackedOnChainConfig` in `types` crate as that will create circular dependency. This depends on DBReader, which depends on the `types`.  The `types` crate is a very thin crate with minimal dependency and probably we need to keep it that way..

Comment (JoshLind) [reply state-sync/inter-component/event-notifications/src/lib.rs:380]: Discussed offline -- my primary concern is that it feels odd for the on chain config logic to be split up. But, will unblock the PR for now and we can add a comment (e.g., for a nice onboarding task 😄).

**Review by JoshLind — APPROVED** [review]
Thanks @sitalkedia!

**Review by zekun000 — APPROVED** [review]

**Comment by sitalkedia** [comment]
@zekun000 - Any thoughts on how we should go about testing it?

**Comment by sitalkedia** [comment]
@JoshLind , @zekun000 - I tested this change by running the smoke test which toggles the shuffling flag and ensures it takes effect. Any other testing we need to do before landing this change?

**Comment by sitalkedia** [comment]
Discussed with @JoshLind and @zekun000 offline, the current testing is good and we can land this change. But as a followup, we might want to add an endpoint to inspect the on-chain config to be able inspect it easily.

**Pull request merged**
